utils/parser.py

```diff
@@ -390,31 +412,32 @@ async def runner(self, line_of_codes: str) -> AsyncGenerator[Tuple[Any], None]:
         for x in (self.iterator, self.executor):
             await x.__anext__()
 
-    async def handle_repl(self, line: str) -> Union[Tuple[str, Exception], str, int]:
+    async def handle_repl(self, line: str) -> Union[Tuple[str, ReplParserDies], int, bool]:
         try:
             return await self.iterator.asend(line)
         except ReplParserDies as e:
-            lines = traceback.format_exception(type(e), e, e.__traceback__)
+            lines = traceback.format_exc()
             return "".join(lines), e
 
-    async def reading_codeblock(self) -> AsyncGenerator[str, None]:
+    async def reading_codeblock(self) -> AsyncIterator[_ExecIterOutput]:
         codes = itertools.dropwhile(lambda x: x == "", self.codeblock.content.splitlines())
         async for line, no, ex in self.runner(codes):
             if isinstance(indent := await self.handle_repl(line), tuple):
                 _, error = indent
                 indicator = ("...", ">>>")[error.mode]
                 yield f"{indicator} {line}"
                 yield indent
+                return
```

Comment:
even though I havent found an execution path that leads to this causing an error, I think it's safer to put return here (or continue, not sure)

Reply:
no, there's a reason why i dont put return in an async generator, remove that